### PR TITLE
mgr/dashboard: Improve level A accessibility for buttons

### PR DESCRIPTION
**Pull request by nSedrickm**
Add titles and aria-labels for icon only buttons so that they can be detected by screen readers
Most of the button accessibility issues come from the datatable component and have been fixed in #46958

image

image

Fixes: https://tracker.ceph.com/issues/56018
Signed-off-by: nsedrickm <nsedrick101@gmail.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/mirroring/overview/overview.component.html [hunk 1]
```diff
@@ -20,7 +20,8 @@
           <div class="input-group-append">
             <button class="btn btn-light"
                     id="editSiteName"
-                    (click)="updateSiteName()">
+                    (click)="updateSiteName()"
+                    [attr.title]="editing ? 'Save' : 'Edit'">
```
Comment (ljflores) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/mirroring/overview/overview.component.html:24]: Interesting, can you explain this syntax?

**Review by ljflores — APPROVED** [review]

**Comment by nSedrickm** [comment]
jenkins test dashboard

**Comment by nSedrickm** [comment]
jenkins test windows

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/mirroring/overview/overview.component.html (hunk 1 above)
Comment (ljflores) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/block/mirroring/overview/overview.component.html:24]: Nice, thanks for explaining!

**Comment by ljflores** [comment]
jenkins test dashboard

**Comment by ljflores** [comment]
jenkins test windows

**Comment by ljflores** [comment]
@epuertat this one's pretty simple; it looks good to me. Do you see anything else that needs to be added?

**Review by nizamial09 — APPROVED** [review]

**Comment by ljflores** [comment]
@epuertat this one is ready to merge-- want to go for it unless you see anything else that needs changing?

**Pull request merged**